src/cdomains/floatDomain.ml

```diff
@@ -37,18 +37,20 @@ module type FloatDomainBase = sig
 
   val of_const : float -> t
 end
-
 module FloatInterval = struct
   type t = (float * float) option [@@deriving eq, ord, to_yojson]
 
   let norm v = 
-    match v with
-    | Some (low, high) -> 
-      if Float.is_finite low && Float.is_finite high then 
-        if low > high then failwith "invalid Interval"
-        else v
-      else None
-    | _ -> None
+    let normed = match v with
+      | Some (low, high) -> 
+        if Float.is_finite low && Float.is_finite high then 
+          if low > high then failwith "invalid Interval"
+          else v
+        else None
+      | _ -> None
+    in if normed == None then 
```

Comment:
If the intention is a comparison with top, use the is_top function instead of manually comparing